runtime/sema/type.go

```diff
@@ -5758,6 +5761,105 @@ func (t *TransactionType) Resolve(_ *TypeParameterTypeOrderedMap) Type {
 	return t
 }
 
+// TransactionRoleType
+
+type TransactionRoleType struct {
+	Members           *StringMemberOrderedMap
+	Fields            []string
+	PrepareParameters []Parameter
+}
+
+var _ Type = &TransactionRoleType{}
+
+func (t *TransactionRoleType) PrepareFunctionType() *FunctionType {
+	return &FunctionType{
+		IsConstructor:        true,
+		Parameters:           t.PrepareParameters,
+		ReturnTypeAnnotation: NewTypeAnnotation(VoidType),
+	}
+}
+
+func (*TransactionRoleType) IsType() {}
+
+func (t *TransactionRoleType) Tag() TypeTag {
+	return TransactionRoleTypeTag
+}
+
+func (*TransactionRoleType) String() string {
+	return "TransactionRole"
+}
+
+func (*TransactionRoleType) QualifiedString() string {
+	return "TransactionRole"
+}
+
+func (*TransactionRoleType) ID() TypeID {
+	return "TransactionRole"
```

Comment:
maybe extract these into a constant

Reply:
Refactored in [a5ca45a](https://github.com/onflow/cadence/pull/2257/commits/a5ca45aba8ec70e8cb41a4bfe7e9dcbd36c6e581)